### ISSUE TITLE
End-to-end Network in Keras

### ISSUE THREAD

**Issue by KlaymenGC** [fchollet/keras#1152]
Hello,

I'm wondering if it's possible to implement a network that takes in an image (RGB images with pixelwise labels) and outputs a probability map of the same size using Keras, I've built the convolutional layers follows by a series of deconvolutional layers and unpooling layers, the output of the final deconvolutional layer will be the same size of the input image, then if I do

```python
model.add(Activation('softmax'))
model.compile(loss='categorical_crossentropy', optimizer='sgd')
```

will it be able to produce a probability map of the image and do the learning?

Any suggestions will be greatly appreciated!

**Comment by phreeza** [comment]
If you use border_mode='same' for all your layers, and the final layer has n_filters = number of labels you want to tag, the output should have the desired shape?

**Comment by KlaymenGC** [comment]
@phreeza thanks for your reply, I've updated the question.

**Comment by shampool** [comment]
＠phreeza what if you end up with a layer with size that can not be divided by the pooling stride.
